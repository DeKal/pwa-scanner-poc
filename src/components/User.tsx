import React, { ReactElement, useState } from 'react'
import PropTypes from 'prop-types'
import CustomerModal from './CustomerModal'
import Button from 'react-bootstrap/Button'
import QrScanner from './QrScanner'
import { UserProps } from 'types/props'
import { fetchStatuses } from 'utils/state'

const User = ({
  user,
  fetchUserStatus,
  fetchUser,
  removeUserSession,
  history,
}: UserProps): ReactElement => {
  const [show, setShow] = useState(false)
  const handleClose = () => setShow(false)
  return (
    <div>
      <h2 className="font-weight-bold text-dark text-center mt-3">
        Foodsavers
      </h2>
      <h5 className="text-center mt-3 mb-3 pl-3 pr-3">
        Scan the QR code of the product for adding to cart
      </h5>
      <QrScanner
        mode="environment"
        setQrCode={(qrCode) => {
          fetchUser(qrCode)
          setShow(true)
        }}
      />
      <Button
        data-test-id="btn-new-session"
        className="mt-5 mb-5 mx-auto w-25"
        variant="outline-dark"
        size="lg"
        block
        onClick={() => {
          removeUserSession()
          history.push('/')
        }}
      >
        New session
      </Button>
      <CustomerModal
        show={show && fetchUserStatus === fetchStatuses.SUCCESS}
        handleClose={handleClose}
        name={`${user.firstName} ${user.lastName}`}
        address={user.address}
        status={user.active}
      />
    </div>
  )
}

export default User
User.defaultProps = {
  user: {},
  fetchUserStatus: null,
  fetchUser: () => {},
}
User.propTypes = {
  user: PropTypes.object,
  fetchUserStatus: PropTypes.string,
  fetchUser: PropTypes.func,
}
