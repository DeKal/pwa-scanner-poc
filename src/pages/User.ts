import { connect } from 'react-redux'
import { withRouter } from 'react-router'

import User from 'components/User'
import { fetchUser } from 'state/actions/user.action'
import { getUser, getFetchUserStatus } from 'state/selectors/user.selector'
import { removeUserSession } from 'state/actions/home.action'

const mapStateToProps = (state) => ({
  user: getUser(state),
  fetchUserStatus: getFetchUserStatus(state),
})

const mapDispatchToProps = {
  fetchUser,
  removeUserSession,
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(User))
