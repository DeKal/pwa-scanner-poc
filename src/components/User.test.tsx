import React from 'react'
import { render } from '@testing-library/react'
import User from './User'

test('renders a User component correctly', () => {
  const { container } = render(<User />)
  expect(container).toMatchSnapshot()
})
