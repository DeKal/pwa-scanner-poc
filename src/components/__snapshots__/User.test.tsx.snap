// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a User component correctly 1`] = `
<div>
  <div>
    <h2
      class="font-weight-bold text-dark text-center mt-3"
    >
      Foodsavers
    </h2>
    <h5
      class="text-center mt-3 mb-3 pl-3 pr-3"
    >
      Scan the QR code of the product for adding to cart
    </h5>
    <div
      class="mockQrScanner"
    >
      {"mode":"environment"}
    </div>
    <button
      class="mt-5 mb-5 mx-auto w-25 btn btn-outline-dark btn-block btn-lg"
      data-test-id="btn-new-session"
      type="button"
    >
      New session
    </button>
    <div
      class="mockCustomerModal"
    >
      {"show":false,"name":"John Cena","address":"My address","status":true}
    </div>
  </div>
</div>
`;
